cpp/lib/Authenticator.hpp

```diff
@@ -87,6 +88,28 @@ namespace Client
     static std::vector<char> SHA256(const std::vector<char> &message);
   };
 
+  class AuthenticatorOKTA : public IAuthenticator
+  {
+  public:
+      AuthenticatorOKTA(SF_CONNECT* conn);
+
+      ~AuthenticatorOKTA();
+
+      void authenticate();
+
+      void updateDataMap(cJSON* dataMap);
```

Comment:
This is a copy of the code in ODBC, except it uses different json library. I don't like the duplication that's happening here Do we have a plan on how to share this functionality between two codebases? @sfc-gh-ext-simba-hx

Reply:
Hello @sfc-gh-jszczerbinski, he is OOO today. As far as I know, we have implemented all the authentications in the libsnowflakeclient and will merge them into the master 2.0.0 libsnowflakeclient first. If we start using the libsnowflakeclient master 2.0, we will refactor the code in ODBC based on the changes in the libsnowflakeclient.

Reply:
@sfc-gh-jszczerbinski Sorry I missed this. Currently libsfclient and odbc use different implementation of rest request and json library and so far no plan to change that. Since authenticator implementation deeply rely on those parts I think we would need some refactoring there to make the authenticator implementation reusable.
It's doable but I think it might be too risky to do large refactoring on the very essential functionalities like rest request and json parser, especially no customer needs on that.

Reply:
@sfc-gh-jszczerbinski Replied your comment in the ticket now I think it's a good idea to use picojson in C++ implementation.
For the rest request part, maybe we could have a virtual function for that to allow overriding on ODBC/libsfclient side, like what I'm doing for bulk updating? https://github.com/snowflakedb/libsnowflakeclient/compare/master...SNOW-710476-bulk-update-essential#diff-1e3b9047a718b31a2af5a7fa63d36f3f4333b7dfa3a0ddde58892d15e015c458R68
@sfc-gh-ext-simba-jy